### PR TITLE
Devhub 3.0.0

### LINKED ISSUE #33: Prevent invalid SSH keys from being submitted to gitolite.
We've had issues with people submitting invalid SSH keys, or SSH keys with exotic algorithms which gitolite doesn't seem to understand.

We should check the validity of SSH keys in some way. I've looked for various implementations of SSH key checkers in Java, but none of them seemed to work. To detect invalid SSH keys on the server (once the damage was already done) I used the following command:

`ssh-keygen -lf <filename>`

which either produces a line like:
`2048 59:78:49:21:35:ea:ba:ba:f4:3c:b8:d8:67:bf:72:1e  <netID> (RSA)`

or not when it's invalid. Perhaps we can use this to validate SSH keys before submitting them to gitolite.

### PR DESCRIPTION
**Pull request by JWGmeligMeyling**
# Changelog
This Pull Request provides the changes required for Devhub to support code review (commenting), tagging and pull requests.

## Features
- Ability to get the `git blame` results for files and diff responses
- Ability to post tags
- Ability to merge branches into the master
- Upgraded to Resteasy `3.0.10.final` and JGit `3.6.1.201501031845`
- Amount of context lines for `diff` requests can be given as a query parameter 
- Files are sent with their `Content-Length` for download progress indication (for this, a `ObjectLoaderBodyWriter` has been introduced)
- Ahead / behind stats are sent with the branches
- HttpClient pool for the Git server

## Deployment notes
- The Git Server has been switched to **Java 8**.
- Mirrors folder might be located elsewhere in deployed environment
- The response models for the `diff` resources are completely changed, because a lot of the data transformations have moved from Devhub to the Git server, as they can be computed more efficiently and with fewer requests.
- The Git Client API has *breaking changes* compared to the previous version because  resources have been separated to different classes.
- Please also merge https://github.com/devhub-tud/Java-Gitolite-Manager/pull/18 to fix #33 

**Because of the breaking changes to the API, this version of the Git server should only be used / deployed in companion with the corresponding Devhub PR**

### CODE REVIEW

**Comment by JWGmeligMeyling** [comment]
Redeployed this branch on the test environment.

**Comment by JWGmeligMeyling** [comment]
The test fails because I have removed the repository and it cant find the Java-Gitolite-Manager.

File: pom.xml [hunk 1]
```diff
@@ -16,38 +16,17 @@
...
-            <snapshots>
-                <enabled>true</enabled>
-                <updatePolicy>always</updatePolicy>
-            </snapshots>
-        </repository>
-    </repositories>
-
 	<dependencies>
 		<dependency>
+			<groupId>javax.ejb</groupId>
```
Comment (michaeldejong) [inline pom.xml]: What are we using this for?

Comment (JWGmeligMeyling) [inline pom.xml]: I believe Jetty accidentally removed it from it's dependencies. We do not use EJB in our own code, but when you remove the dependency, the server fails to start.

Comment (JWGmeligMeyling) [inline pom.xml]: It has to do with moving to `<resteasy.version>3.0.10.Final</resteasy.version>`

**Pull request merged**